lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -185,13 +195,16 @@ void ExecutionEngine::insertCompiledFunction(
   functionMap[name] = func.get();
   compiledFunctions_[name] = std::move(func);
 
-  std::promise<llvm::Error> addPromise;
-  auto fut = addPromise.get_future();
+  llvm::Error err = MAKE_ERR("Function did not run");
+
   device_->addNetwork(&M_, std::move(functionMap),
-                      [&addPromise](const Module *, llvm::Error err) {
-                        addPromise.set_value(std::move(err));
+                      [this, &err](const Module *, llvm::Error err_) {
+                        err = std::move(err_);
+                        cv_.notify_one();
                       });
-  EXIT_ON_ERR(fut.get());
+  std::unique_lock<std::mutex> l(m_);
+  cv_.wait_for(l, 2s);
```

Comment:
What is the point of the wait for 2s?

Reply:
Stops the test from deadlocking if the function doesn't get called.